optimum/habana/transformers/models/llama/modeling_llama.py

```diff
@@ -534,27 +549,54 @@ def pre_attn_forward(
 
             softmax_mode = "fast" if flash_attention_fast_softmax else "None"
 
+            global flash_attention_in_fp8
+            if flash_attention_in_fp8 is True:
+                query_states, key_states, value_states, attention_mask = gaudi_llama_repeat_kv(
+                    query_states,
+                    key_states,
+                    value_states,
+                    attention_mask,
+                    self.num_key_value_groups,
+                )
+
             if q_len == 1:
                 # next token
                 use_recompute = True if os.getenv("QUANT_CONFIG", "") else False
                 with ht.sdp_kernel(enable_recompute=use_recompute):
-                    attn_output = self.fused_scaled_dot_product_attention(
-                        query_states, key_states, value_states, attention_mask, 0.0, False, None, 'None'
-                    )
+                    if flash_attention_in_fp8 is True:
+                        attn_output = self.fused_scaled_dot_product_attention(
+                            query_states, key_states, value_states, attention_mask, 0.0, False, None, softmax_mode
```

Comment:
Original code did not have "softmax_mode" as last argument, also this case looks like something which will be applicable only for inference. Please check and remove the change if possible, we want to leave inference path untouched (as much as possible) to avoid regressions.

Reply:
It had the argument with "None" which will also be filled with the same value (see: L550)

Reply:
Do we really need to add change for flash_attention_in_fp8 in this part of the loop? "q_len == 1" is usually only applicable for inference decode part whereas we are adding TE for training only.

Reply:
Fixed (and hardcoded sotmax_mode = "None")

Reply:
@pbielak I think we need to keep softmax_mode, basically change nothing w.r.t. base code to avoid regression in inference cases. Please call me if comment is not clear :-)